### ISSUE TITLE
[Feature Request] Compare current model to most recent (or other) commit

### ISSUE THREAD

**Issue by michaelansley** [archimatetool/archi-modelrepository-plugin#123]
Hi,

I'd like to propose a new feature, which allows a user to compare the current model (assumed to have changed since the last commit) to the last commit, or potentially some other commit.

Such a function could be accessed by a simple menu item on the Collaboration menu, or context menu item for the Change History UI.

Implementation could be similar to the diff management UI, with some minor changes.

Regards,
Michael

**Comment by Phillipus** [comment]
@jbsarrodie and I discussed this recently to at least show some diff when doing a commit.

**Comment by michaelansley** [comment]
Great!  I often find myself wondering what, exactly, I'm about to commit when I, for example, switch branches.
Or wondering what exactly I'm committing, compared to what I think I'm committing.  Given that I'm usually working on three or four different aspects of the model at any one time, I can easily do work in the wrong branch, especially if it is small changes to demonstrate something to a colleague.

**Comment by Phillipus** [comment]
Yes, indeed. We discussed this same thing and it did seem essential to want to see what changed in the same way as when you do a merge conflict.

**Comment by Phillipus** [comment]
Yes, indeed. We discussed this same thing and it did seem essential to want to see what changed in the same way as when you do a merge conflict.

**Comment by WatchTh1** [comment]
Will this feature be put into roadmap ?

**Comment by Phillipus** [comment]
> Will this feature be put into roadmap ?

Yes. But don't know when I will work on it (I'm working on a lot of stuff)

**Comment by eduardobbs** [comment]
Hi, if it helps I've created a JArchi script and a batch that does the job for me on Windows.
This is the content of the JArchi script:
`exec("GitCompare.bat");
console.log("GitK will open, the comparison will also be available on Archi under the Scripts Manager tab: 'GitCompareRESULT.diff' file");`

And this is the content of the batch file:
`cd instance\model-repository\architecture_ecommerce_model
git show > jarchi-scripts\Architecture\GitCompareRESULT.diff
gitk &`

I use a portable version and I point my repository folder resides on a relative path to Archi, but you could set it up differently, here is my folder structure:
![image](https://user-images.githubusercontent.com/5373152/116605290-22f17400-a927-11eb-8e65-cdbe0c8b7d65.png)

**Comment by CamilleDominique** [comment]
Hi all,
For me the issue is not only to know in advance what are the differences between current local save and last commit on an xml point of view. An xml diff would be nice already but in reality not really useful if there are a lot of differences and hundreds of xml files have been changed.
Something even more useful would be to sort the differences by views and provide visual comparison to the user.
That is a personal idea, not sure how feasible it is though

**Comment by Phillipus** [comment]
> Has this feature request been included in 2022/2023 roadmap?

It's something we're looking at for coArchi version 2.

**Comment by startsevsa** [comment]
> > Has this feature request been included in 2022/2023 roadmap?
> 
> It's something we're looking at for coArchi version 2.

Maybe start from parsing the result of the _git ls-files -m_ command ?

**Comment by Phillipus** [comment]
coArchi2
